lib/package/dynamodb/compileMethodsToDynamodb.js

```diff
@@ -0,0 +1,387 @@
+        'application/x-www-form-urlencoded': this.getGetItemDefaultDynamodbResponseTemplate()
+      }
+    }
+
+    return {}
+  },
+
+  getGetItemDefaultDynamodbResponseTemplate() {
+    return `#set($item = $input.path('$.Item')){#foreach($key in $item.keySet())#set ($value = $item.get($key))#foreach( $type in $value.keySet())"$key":"$value.get($type)"#if($foreach.hasNext()),#end#end#if($foreach.hasNext()),#end#end}`
+  },
+
+  getPutItemDefaultDynamodbResponseTemplate(http) {
+    const objectRequestParam = this.getDynamodbObjectHashkeyParameter(http)
+    return {
+      'Fn::Sub': [
+        '{"${HashKey}": "${HashAttributeValue}"}',
+        {
+          HashKey: `${objectRequestParam.key}`,
+          HashAttributeValue: `${objectRequestParam.attributeValue}`
+        }
+      ]
+    }
+  },
+
+  buildDefaultDynamodbRequestTemplate(http) {
+    switch (http.action) {
+      case 'PutItem':
+        return this.buildDefaultDynamodbPutItemRequestTemplate(http)
+      case 'GetItem':
+        return this.buildDefaultDynamodbGetItemRequestTemplate(http)
+      case 'DeleteItem':
+        return this.buildDefaultDynamodbDeleteItemRequestTemplate(http)
+      case 'UpdateItem':
+        return this.buildDefaultDynamodbUpdateItemRequestTemplate(http)
+    }
+  },
+
+  buildDefaultDynamodbDeleteItemRequestTemplate(http) {
+    const fuSubValues = {
+      TableName: http.tableName
+    }
+
+    let requestTemplate = '{"TableName": "${TableName}","Key":{'
+    if (_.has(http, 'hashKey')) {
+      requestTemplate += '"${HashKey}": {"${HashAttributeType}": "${HashAttributeValue}"}'
+      Object.assign(fuSubValues, this.getDynamodbHashkeyFnSubValues(http))
+    }
+
+    if (_.has(http, 'rangeKey')) {
+      requestTemplate += ',"${RangeKey}": {"${RangeAttributeType}": "${RangeAttributeValue}"}'
+      Object.assign(fuSubValues, this.getDynamodbRangekeyFnSubValues(http))
+    }
+    requestTemplate += '}'
+    if (_.has(http, 'condition')) {
+      requestTemplate += ',"ConditionExpression": "${ConditionExpression}"'
+      fuSubValues['ConditionExpression'] = http.condition
+    }
+    requestTemplate += '}'
+    return {
+      'Fn::Sub': [`${requestTemplate}`, fuSubValues]
+    }
+  },
+
+  buildDefaultDynamodbUpdateItemRequestTemplate(http) {
+    const fuSubValues = {
+      TableName: http.tableName
+    }
+
+    let requestTemplate = '{"TableName": "${TableName}","Key":{'
+    if (_.has(http, 'hashKey')) {
+      requestTemplate += '"${HashKey}": {"${HashAttributeType}": "${HashAttributeValue}"}'
+      Object.assign(fuSubValues, this.getDynamodbHashkeyFnSubValues(http))
+    }
+
+    if (_.has(http, 'rangeKey')) {
+      requestTemplate += ',"${RangeKey}": {"${RangeAttributeType}": "${RangeAttributeValue}"}'
+      Object.assign(fuSubValues, this.getDynamodbRangekeyFnSubValues(http))
+    }
+
+    requestTemplate += '}'
+    requestTemplate += `
+      #set ($body = $input.path('$'))
```

Comment:
Should we inject the update expression directly from the client request?
This feels unsafe.

I'm not sure, but it might be better to have `http.updateExpression`, `http.expressionAttributeValues` and `http.expressionAttributeNames`.